server/matchmaker/algorithm.py

```diff
@@ -52,13 +54,37 @@ def find(self) -> List[Match]:
                     search, preferred, search.quality_with(preferred),
                     search.match_threshold, preferred.match_threshold
                 )
-                if not search.matches_with(preferred):
-                    continue
 
                 self._propose(search, preferred)
 
+        self._forcefully_match_unmatched_newbies()
+
         return self._remove_duplicates()
 
+    def _forcefully_match_unmatched_newbies(self):
```

Comment:
Is it practical to move this out of the `StableMarriage` class? This isn't part of the stable marriage implementation any more. I had refactored to a class to cut down on the explicit parameter passing, but maybe the class should be renamed?

Reply:
These are some slightly bigger changes, so could you tell me what you think about this https://github.com/cleborys/server/pull/5 before I include them in the PR? :)

Reply:
Oh yea that looks pretty nice!